### PR TITLE
Travis: use --work-tree with git instead of `cd`

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by psychon** [comment]
This is just for aesthetics, right? Since it seems to work (does it actually?), it's fine with me.

**Comment by blueyed** [comment]
Well.. it triggered pushing of the docs after all (build: https://travis-ci.com/awesomeWM/awesome-www/builds/102143617): https://github.com/awesomeWM/awesomeWM.github.io/commit/02dbbaa32beeeccb3336e9706b5ed2410efb591b (which is not cool)
(investigating)

**Comment by blueyed** [comment]
Should have used `--git-dir` there..

**Comment by blueyed** [comment]
Amended.
It is mainly meant to protect against ending in some unexpected dir.

File: .travis.yml [hunk 1]
```diff
@@ -49,18 +49,19 @@ before_install:
...
-        branch="$TRAVIS_BRANCH"
-      else
-        branch="pr-$TRAVIS_PULL_REQUEST"
-      fi
-      if [ "$branch" != master ]; then
-        cd $BUILD_WEB
-        if ! git checkout -b "$branch" "origin/$branch"; then
-          git checkout -b "$branch"
+      (
+        cd "$BUILD_WEB"
```
Comment (actionless) [inline .travis.yml]: but it's still `cd`, not `--git-dir/--work-tree/-C/whatever`

Comment (blueyed) [reply .travis.yml]: It's in a subshell now.

**Comment by actionless** [comment]
> Well.. it triggered pushing of the docs after all

oops, i've missed your message

i hotfixed that as a critical issue:
https://github.com/awesomeWM/awesome-www/commit/3e0aaa5050c830ed4e107084df95a0fc3a2735d7

i recommend reviewing the commit log of https://github.com/awesomeWM/awesomeWM.github.io for unexpected changes there

**Comment by Elv13** [comment]
Obsoleted by #170

**Pull request closed without merging**